### PR TITLE
Adding webhook docs

### PR DESCRIPTION
**Pull request by nitro-neal**
# Overview
Added documentation in on how to use Webhooks in the SSI-Service

### CODE REVIEW

File: doc/WEBHOOK.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+````json
+POST - http://localhost:8080/v1/webhooks
+{
+    "noun": "DID",
+    "verb": "Create",
+    "url": "http://my-service-that-recieves-webhooks.com/webhook"
+}
+````
+
+This command creates a webhook inside of the SSI-Service. It basically tells the SSI-Service that when a did is created post a notification to the specified URL (http://my-service-that-recieves-webhooks.com/webhook)
```
Comment (andresuribe87) [inline doc/WEBHOOK.md:60]:
```suggestion
This command creates a webhook inside of the SSI-Service. It tells the SSI-Service to make a POST request to the specified URL (http://my-service-that-recieves-webhooks.com/webhook) after a DID is created (i.e. after the [Create DID Document](https://developer.tbd.website/docs/apis/ssi-service#tag/WebhookAPI) endpoint is called).
```

File: doc/WEBHOOK.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        [
+          "#z6MkqE3sxyrKYS2UaHn7tpQz66NHaSfcuLrBuVtiWHeRoMtB"
+        ]
+      ]
+    }
+  }
+}
+````
+
+This response object has the Noun and Verb that happened that fired it, and the data attached to the response of the initial call
```
Comment (andresuribe87) [inline doc/WEBHOOK.md:106]:
```suggestion
This response object has the Noun and Verb that happened that fired it, and the data attached to the response of the initial call.
```

File: doc/WEBHOOK.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+````json
+POST - http://localhost:8080/v1/webhooks
+{
+    "noun": "Submission",
+    "verb": "Create",
+    "url": "http://my-service-that-recieves-webhooks.com/webhook"
+}
+````
+
+Use the SSI Service to set up a Presentation Definition and corresponding Presentation Request to accept a KYC credential. We have the prerequisites for this in our steelthread integration test
```
Comment (andresuribe87) [inline doc/WEBHOOK.md:123]: Link to the integration test?

File: doc/WEBHOOK.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+When someone uses the SSI-Service to Create a submission:
+````json
+POST - http://localhost:8080/v1/presentations/submissions
+{
+  "submissionJwt": "..."
+}
+````
+
+A new submission is created the service that is listening to the webhook will recieve this data:
```
Comment (andresuribe87) [inline doc/WEBHOOK.md:133]: I'm not sure I understood this phrase, mind clarifying?

**Review by andresuribe87 — COMMENTED** [review]
This is awesome! I think it would be easier to discover by devs if it's in the official API reference (https://developer.tbd.website/docs/apis/ssi-service#tag/WebhookAPI). Would it make sense to include it there?

**Comment by decentralgabe** [comment]
> This is awesome! I think it would be easier to discover by devs if it's in the official API reference (https://developer.tbd.website/docs/apis/ssi-service#tag/WebhookAPI). Would it make sense to include it there?

I agree. I think *both* would be better if we could achieve it. I like the idea of having living docs in the `docs` repo, but making sure any APIs are properly documented too. Sometimes API docs aren't the right place to go into more detail about how things in the system work

File: README.md [hunk 5]
```diff
@@ -162,6 +162,14 @@ The project follows a proposal-based improvement format called [SIPs, outlined h
 Please [join Discord](https://discord.com/invite/tbd),
 or open an [issue](https://github.com/TBD54566975/ssi-service/issues) if you are interested in helping shape the future of the project.
 
+## Project Components References
+
+| Resource                                                                        | Description                                                                   |
+|---------------------------------------------------------------------------------|-------------------------------------------------------------------------------|
+| [Webhooks](https://github.com/TBD54566975/ssi-service/blob/main/doc/WEBHOOK.md) | Describes how to use webhooks in the SSI Service                              |
```
Comment (decentralgabe) [inline README.md:169]: could you change this to point to a generic README in the `docs` folder, which has a similar table, which includes `Webhooks` as the first entry?

File: doc/WEBHOOK.md [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+{
+    "noun": "DID",
+    "verb": "Create",
+    "url": "http://my-service-that-recieves-webhooks.com/webhook"
+}
+````
+
+This command creates a webhook inside of the SSI-Service. It basically tells the SSI-Service that when a did is created post a notification to the specified URL (http://my-service-that-recieves-webhooks.com/webhook)
+
+When a new did is created the service that is listening to the webhook will recieve this data:
```
Comment (decentralgabe) [inline doc/WEBHOOK.md:62]:
```suggestion
When a new DID is created the service that is listening for the webhook will receive this data:
```

File: doc/WEBHOOK.md [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+    }
+  }
+}
+````
+
+This response object has the Noun and Verb that happened that fired it, and the data attached to the response of the initial call
+
+
+# Presentation Exchange Webhook Example
+Here is an example of how to setup a webhook to fire when a new presentation submission is created:
```
Comment (decentralgabe) [inline doc/WEBHOOK.md:110]:
```suggestion
Here is an example of how to setup a webhook to fire when a new presentation submission is received by the service:
```

File: doc/WEBHOOK.md [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+
+
+First setup a webhook object in the SSI-Service that will POST to the given URL when a Submission is created
```
Comment (decentralgabe) [inline doc/WEBHOOK.md:113]:
```suggestion
First setup a webhook object in the SSI-Service that will POST to the given URL when a Submission is received by the service:
```

Comment (decentralgabe) [inline doc/WEBHOOK.md:113]:
```suggestion
First setup a webhook object in the SSI-Service that will POST to the given URL when a Submission is received by the service:
```

File: doc/WEBHOOK.md (hunk 4 above)
Comment (decentralgabe) [reply doc/WEBHOOK.md:133]:
```suggestion
Upon receiving a new submission the service that is registered to listen for the webhook will receive this data:
```

File: doc/WEBHOOK.md [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+  "url": "http://my-service-that-recieves-webhooks.com/webhook",
+  "data": {
+    "id": "presentations/submissions/e875b34e-35fd-4ad9-8805-4f16bf98df71",
+    "done": false,
+    "result": {}
+  }
+} 
+````
+
+The external webhook service that received this notification can then get the required objects, do validation, and make a call to the SSI-Service to approve or to not approve.
```
Comment (decentralgabe) [inline doc/WEBHOOK.md:147]: might be good to show an example of an approval

Comment (decentralgabe) [reply doc/WEBHOOK.md:147]: or link to a blank doc where that approval will be (assuming we'll need docs for operations)

File: doc/WEBHOOK.md (hunk 3 above)
Comment (nitro-neal) [reply doc/WEBHOOK.md:123]: LOL an I'll put a link to our streelthread integ test in the readme too

Comment (andresuribe87) [reply doc/WEBHOOK.md]: You could start a test http server within golang. See https://pkg.go.dev/net/http/httptest#NewServer. That might be the easiest.

**Comment by decentralgabe** [comment]
@nitro-neal sorry one more change. would you mind deleting `swagger.yaml` from the `docs` folder and adding a link in the new README to the [swagger branch](https://github.com/TBD54566975/ssi-service/tree/main_swagger) with a note on viewing the docs by running the container?

File: integration/presentation_exchange_integration_test.go [hunk 10]
```diff
@@ -15,7 +15,7 @@ var presentationExchangeContext = NewTestContext("PresentationExchange")
 
 func TestCreateParticipants(t *testing.T) {
 	if testing.Short() {
-		t.Skip("skipping integration test")
+		t.Skip("skipping integration testz")
```
Comment (decentralgabe) [inline integration/presentation_exchange_integration_test.go:18]: ?

**Review by decentralgabe — APPROVED** [review]

**Pull request merged**